### PR TITLE
api: Refactor Service Middleware

### LINKED ISSUE #4846: Service Middleware doesn't put `next` handler into middleware actually
### What did you do?
I want to get the HTTP duration of `GetRegions`from Prometheus audit.

### What did you expect to see?
If regions is up to 10K, it will take hundreds of ms.

### What did you see instead?
It will take a few ms.

https://github.com/tikv/pd/blob/1fc68b2abd664e6bf73bdda874399c8db4435cf6/server/api/router.go#L103-L108
Audit backends whose `ProcessBeforeHandler()` is false will process before handler processing not after.

It will also affect Rate-Limit Middleware.

### What version of PD are you using (`pd-server -V`)?
master

### PR DESCRIPTION
**Pull request by CabinfeverB**
Signed-off-by: Cabinfever_B <cabinfeveroier@gmail.com>

### What problem does this PR solve?

Issue Number: Close #4846 Ref 

### What is changed and how it works?

Create one middleware for each HTTP API interface.
And I remove failpoint in audit middleware because it has two backends to test. 

```commit-message
Refactor Service Middleware
```

### Check List

Tests

- Unit test
- Integration test

### Release note

```release-note
None.
```

### CODE REVIEW

File: server/api/middleware.go [hunk 1]
```diff
@@ -106,23 +121,15 @@ func (s *auditMiddleware) ServeHTTP(w http.ResponseWriter, r *http.Request, next
...
-	requestInfo, ok := requestutil.RequestInfoFrom(r.Context())
-	if !ok {
-		log.Error("failed to get request info when auditing")
-		next(w, r)
-		return
-	}
+	// There is no need to check whether requestInfo is available like getting RaftCluster
+	requestInfo, _ := requestutil.RequestInfoFrom(r.Context())
 
 	labels := s.svr.GetServiceAuditBackendLabels(requestInfo.ServiceLabel)
```
Comment (rleungx) [inline server/api/middleware.go:127]: If there is no requestInfo, we don't need to get it?

Comment (CabinfeverB) [reply server/api/middleware.go:127]: > If there is no requestInfo, we don't need to get it?

If we open audit or rate-limit switch, there must be a requestInfo.

Comment (rleungx) [reply server/api/middleware.go:127]: Please update the comment here.

**Review by rleungx — APPROVED** [review]
The rest LGTM.

**Review by nolouch — APPROVED** [review]

**Pull request merged**